config/dynamic.c

```diff
@@ -601,6 +605,21 @@ run_dynamic_config(struct gk_config *gk_conf, struct gt_config *gt_conf,
 	if (ret < 0)
 		goto put_gk_gt_config;
 
+	ret = chown(dy_conf->server_path, net_conf->pw_uid, net_conf->pw_gid);
```

Comment:
1. Only call `chown()` when `net_conf->pw_uid != 0`.

2. Can't we call `fchown()` on `dy_conf->sock_fd` instead of calling `chown()`? This avoids rare file-system-related race conditions.